tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/DisableTestsTestCase.java

```diff
@@ -39,10 +40,10 @@ public void setup() throws BallerinaTestException {
 
     @Test
     public void testDisablingTests() throws BallerinaTestException {
-        String msg = "2 passing";
-        LogLeecher clientLeecher = new LogLeecher(msg);
-        balClient.runMain("test", new String[]{"disable-test.bal"}, null, new String[]{},
-                new LogLeecher[]{clientLeecher}, projectPath);
-        clientLeecher.waitForText(20000);
+        String errorOutput = balClient.runMainAndReadStdOut("test", new String[]{"disable-test.bal"},
+                new HashMap<>(), projectPath, true);
+        if (errorOutput.contains("error: there are test failures")) {
+            throw new BallerinaTestException("Test failed due to disable test failure in test framework");
+        }
```

Comment:
Since I've noticed this same logic is being duplicated in many places, how about defining an assertion util as follows? WDYT? 

```java
private void assertForTestFailures(String programOutput, String errorMessage) throws BallerinaTestException {
    if (errorOutput.contains("error: there are test failures")) {
         throw new BallerinaTestException(errMessage);
    }
}

Reply:
Fixed with 8aae1b9